From 1a3167719746a6a663430a3da8bb5c85ae59df65 Mon Sep 17 00:00:00 2001
From: Zhang Yiqun <zhangyiqun@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:06 +0800
Subject: [PATCH 073/150] dt-bindings: Add bindings for Phytium I2S controller

This patch documents the DT binding for the Phytium I2S controller.

Signed-off-by: Zhang Yiqun <zhangyiqun@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Id63996cdb0e2681e370c077779b633e2d7ce0555
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/sound/phytium,i2s.yaml           | 52 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 53 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/phytium,i2s.yaml

diff --git a/Documentation/devicetree/bindings/sound/phytium,i2s.yaml b/Documentation/devicetree/bindings/sound/phytium,i2s.yaml
new file mode 100644
index 000000000000..06e5613cdbcf
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/phytium,i2s.yaml
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/phytium,i2s.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium I2S controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,i2s
+
+  reg:
+    maxItems: 2
+    description: |
+      It contains two register region. The first one is for physical base
+      address and length of I2S controller. The second one is for physical
+      base address and length of DMA_BDL controller.
+
+  interrupts:
+    description: |
+      The interrupt line number for the I2S controller. It should contain
+      the DMA_BDL interrupt.
+    maxItems: 1
+
+  clocks:
+    description: Sampling rate reference clock
+    maxItems: 1
+
+  dai-name:
+    $ref: /schemas/types.yaml#/definitions/string
+    description: it will set dai's name used in driver.
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - dai-name
+
+examples:
+  - |
+    i2s@28009000 {
+      compatible = "phytium,i2s";
+      reg = <0x0 0x28009000 0x0 0x1000>, <0x0 0x28005000 0x0 0x1000>;
+      interrupts = <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
+      clocks = <&sysclk_600mhz>;
+      dai-name = "phytium-i2s-lsd";
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 5af0776d6c87..0c3b48347495 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2551,6 +2551,7 @@ F:	Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml
 F:	Documentation/devicetree/bindings/pwm/phytium,pwm.yaml
 F:	Documentation/devicetree/bindings/rng/phytium,rng.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
+F:	Documentation/devicetree/bindings/sound/phytium,i2s.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,spi.yaml
 F:	Documentation/devicetree/bindings/usb/phytium,usb2.yaml
-- 
2.47.0

